File: PageObjectJava/src/test/java/PageObjects/AutomationPracticeStore/ErrorMessageException.java
package PageObjects.AutomationPracticeStore;

import org.openqa.selenium.WebDriverException;

import java.util.ArrayList;
import java.util.List;

public class ErrorMessageException extends WebDriverException {

    private final List<String> errors;

    public ErrorMessageException(String message) {
        super(message);
        this.errors = new ArrayList<>();
        this.errors.add(message);
    }

    public ErrorMessageException(List<String> errorsList){
        super(messageBuilder(errorsList));
        this.errors = errorsList;
    }

    public List<String> getErrorsList(){
        return this.errors;
    }

    private static String messageBuilder(List<String> errors){
        StringBuilder builder = new StringBuilder();
        for (String error : errors)
            builder.append(error + "\n");
        return builder.toString();
    }
}

File: PageObjectDotNet/AutomationPractice/PageObjects/AutomationPracticeStore/Objects/ErrorMessageException.cs
using OpenQA.Selenium;
using System;
using System.Collections.Generic;
using System.Linq;

namespace AutomationPractice.PageObjects.AutomationPracticeStore.Objects
{
    [Serializable]
    internal class ErrorMessageException : WebDriverException
    {
        public IList<string> ErrorList { get; }

        public ErrorMessageException(string message) : base(message) { }

        public ErrorMessageException(IList<string> errorList) : base(MessageBuilder(errorList)) { ErrorList = errorList; }

        public ErrorMessageException(string message, Exception innerException) : base(message, innerException) { }


        private static string MessageBuilder(IList<string> errors)
        {
            string errorMsg = string.Empty;
            errors.ToList().ForEach(error =>
            {
                errorMsg = string.Concat(errorMsg, error + "\n");
            });
            return errorMsg;
        }

       
    }
}